sympy/stats/drv_types.py

```diff
@@ -90,7 +90,8 @@ class GeometricDistribution(SingleDiscreteDistribution):
 
     @staticmethod
     def check(p):
-        _value_check(0 < p and p <= 1, "p must be between 0 and 1")
```

Comment:
Instead of splitting in two lines, would this be better as `_value_check(And(0 < p, p <= 1), "p must be between 0 and 1")`? That is, replacing Python's `and` with SymPy's And.

Reply:
Thank you, this seems better. I'll update the PR.